src/actions/OpenFileAction.java

```diff
@@ -0,0 +1,48 @@
+package actions;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import utilities.RecentFilesManager;
+
+public class OpenFileAction extends TextAction {
+    private String filePath;
+    private RecentFilesManager recentFilesManager;
+
+    // For opening a file using new file menu
+    public OpenFileAction(JTextPane textPane, RecentFilesManager recentFilesManager) {
```

Comment:
Having two constructors handle open file and open file via recent items is very elegant.